plugins/obs-outputs/rtmp-stream.c

```diff
@@ -31,7 +31,15 @@
 
 /* dynamic bitrate coefficients */
 #define DBR_INC_TIMER (30ULL * SEC_TO_NSEC)
-#define DBR_TRIGGER_USEC (200ULL * MSEC_TO_USEC)
+#define DBR_LOW_NORMAL_TIMER (400ULL * MSEC_TO_NSEC)
+#define DBR_LOW_HIGH_TIMER (600ULL * MSEC_TO_NSEC)
+#define DBR_LOW_SEVERE_TIMER (1000ULL * MSEC_TO_NSEC)
+#define DBR_LOW_NORMAL_RATE 50
+#define DBR_LOW_HIGH_RATE 20
+#define DBR_LOW_SEVERE_RATE 10
+#define DBR_NORMAL_TRIGGER_USEC (200ULL * MSEC_TO_USEC)
+#define DBR_HIGH_TRIGGER_USEC (400ULL * MSEC_TO_USEC)
+#define DBR_SEVERE_TRIGGER_USEC (700ULL * MSEC_TO_USEC)
```

Comment:
All these names strike me as very odd, would love an explanation on what they mean

Reply:
The lowering of bitrate is based on the time of the buffered encoder packets waiting to be sent, so I changed the trigger to lower the bitrate to this:
- Normal trigger is 200 ms and it lowers the bitrate by 2% of the original
- High trigger is 400 ms and it lowers the bitrate by 5% of the original
- Severe trigger is 700 ms and it lowers the bitrate by 10% of the original

Also added timeouts for each trigger to not let the bitrate lower that often if the time of the buffered encoder packets waiting to be sent stay on that range:
- Normal trigger has timeout of 400 ms
- High trigger has timeout of 600 ms
- Severe trigger has timeout of 1 second

Reply:
Got it, thanks for the explanation. All these names and this data structure is very not intuitive, I would rework it.

Reply:
I will change the names, but what do you mean about data structure?
If it is the actual value of these coefficients I just followed the same pattern that was being used in the code there

Reply:
Changed the names

Reply:
I would put in an enum severity {LOW, NORMAL, HIGH} then use a different maps for trigger, timer and rate and then associate each coefficient with the specific severity.

Reply:
So create severity enum, then create 3 multidimensional arrays (`int trigger[3][1], int timer [3][1] and int rate[3][1]`)then do something like this for example `int trigger[3][1] = {{LOW, 50}, {NORMAL, 20}, {HIGH, 10}}`?

Reply:
Forget about the above, how do I do the map and associate part on regular C without having to create a map structure from scratch?

Reply:
Many options I think, I would maybe use 2 dimensional arrays to solve this problem.